pkg/daemon/ceph/agent/flexvolume/controller.go

```diff
@@ -347,8 +362,36 @@ func (c *Controller) GetClientAccessInfo(clusterNamespace string, clientAccessIn
 	}
 
 	clientAccessInfo.MonAddresses = monEndpoints
-	clientAccessInfo.SecretKey = clusterInfo.AdminSecret
-	clientAccessInfo.UserName = "admin"
+
+	clientAccessInfo.UserName = args[2]
+	clientAccessInfo.SecretKey = args[3]
+
+	if c.mountSecurityMode == agent.MountSecurityModeRestricted && (clientAccessInfo.UserName == "" || clientAccessInfo.SecretKey == "") {
+		return fmt.Errorf("no mount user and/or mount secret given")
+	}
+
+	if c.mountSecurityMode == agent.MountSecurityModeAny && clientAccessInfo.UserName == "" {
+		clientAccessInfo.UserName = "admin"
```

Comment:
Instead of requiring the mountSecurityMode, can't the default be `admin` and then be overridden by the storage class settings? It still seems we don't need the global mountSecurityMode setting

Reply:
Yes, the mountSecurityMode is needed because if an user would simply not set `mountUser` in a volume (e.g., kube-registry example) the user tries to mount in a Deployment they would then be able to use `admin` because it is the default.